src/decompress.rs

```diff
@@ -494,6 +541,43 @@ impl<'src> DecompressStarted<'src> {
         Some(image_dst)
     }
 
+    /// Supports any pixel type that is marked as "plain old data", see bytemuck crate.
+    /// `[u8; 3]` and `rgb::RGB8` are fine, for example.
+    /// Allocation-less version of `read_scanlines`
+    #[track_caller]
+    pub fn read_scanlines_into<T: rgb::Pod>(&mut self, dest: &mut [T]) -> usize {
+        let num_components = self.color_space().num_components();
+        // assert_eq!(num_components, mem::size_of::<T>());
+        let width = self.width();
+        let height = self.height();
+        assert_eq!(height * width * num_components, dest.len());
```

Comment:
I just realized that this function actually expects a flat buffer (`[u8]` made up of `rgbrgb` etc) which isn't consistent with the `read_scanlines` api, which writes into what basically boils down to a `&[[u8;3]]`.

This **will** cause confusion. I'd better make it compliant with the existing API.

I'll adjust it to follow, but I'll need to test it again.